core/services/synchronization/telemetry_ingress_batch_client.go

```diff
@@ -0,0 +1,301 @@
+package synchronization
+
+import (
+	"context"
+	"errors"
+	"net/url"
+	"sync"
+	"time"
+
+	"go.uber.org/atomic"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services"
+	"github.com/smartcontractkit/chainlink/core/services/keystore"
+	telemPb "github.com/smartcontractkit/chainlink/core/services/synchronization/telem"
+	"github.com/smartcontractkit/chainlink/core/utils"
+	"github.com/smartcontractkit/wsrpc"
+	"github.com/smartcontractkit/wsrpc/examples/simple/keys"
+)
+
+//go:generate mockery --dir ./telem --name TelemClient --output ./mocks/ --case=underscore
+
+// TelemetryIngressBatchClient encapsulates all the functionality needed to
+// send telemetry to the ingress server using wsrpc
+type TelemetryIngressBatchClient interface {
+	services.Service
+	Start() error
+	Close() error
+	Send(TelemPayload)
+}
+
+// NoopTelemetryIngressBatchClient is a no-op interface for TelemetryIngressBatchClient
+type NoopTelemetryIngressBatchClient struct{}
+
+// Start is a no-op
+func (NoopTelemetryIngressBatchClient) Start() error { return nil }
+
+// Close is a no-op
+func (NoopTelemetryIngressBatchClient) Close() error { return nil }
+
+// Send is a no-op
+func (NoopTelemetryIngressBatchClient) Send(TelemPayload) {}
+
+// Healthy is a no-op
+func (NoopTelemetryIngressBatchClient) Healthy() error { return nil }
```

Comment:
Is this needed?
Don't see interface or actual struct have this.

Same with Ready() method

Reply:
Looks like it's required because the the noop struct needs to implement the same interface as the batch client when it's wired up in `application.go`. Both `Ready` and `Healthy` are defined on `Service`, which the batch client implements